GDJS/Runtime/spriteruntimeobject.js

```diff
@@ -942,125 +1040,345 @@ gdjs.SpriteRuntimeObject.prototype.isFlippedY = function() {
+        gl_FragColor = final;
+    }
+`
+    .split('\n')
+    .reduce((c, a) => c + a.trim() + '\n');
+
+  var filter = new PIXI.Filter(vertexSrc, fragSrc, uniforms);
+
+  filter.apply = function (filterManager, input, output) {
+    //this.uniforms = uniforms;
+
+    sprite_sc.texture.transform.update();
+    var sprite_scMatrix = new PIXI.Matrix();
+
+    this.uniforms.texture_sc = sprite_sc.texture;
+    this.uniforms.otherMatrix = filterManager
+      .calculateSpriteMatrix(sprite_scMatrix, sprite_sc)
+      .prepend(sprite_sc.texture.transform.mapCoord);
+
+    // draw the filter...
+    filterManager.applyFilter(this, input, output);
+  };
+
+  this._renderer.setFilter(filter);
+};
+
+/**
+ * Pick a color on sprite and apply a mask on it for make color transparent.
+ * @param {string} rgbColor Color to make transparent, in RGB format ("128;200;255").
+ */
+gdjs.SpriteRuntimeObject.prototype.makeColorTransparent = function (
+  rgbColor,
+  threshold
+) {
+  //https://jsfiddle.net/Babar31/vqyk8gmf/4/
+
+  //Convert rgbColor "128;200;255" in [128, 200, 255]
+  var rgbColor = rgbColor.split(';');
+
+  // create filter
+  var uniforms = {
+    u_colorToRemove: [
+      rgbColor[0] / 255.0,
+      rgbColor[1] / 255.0,
+      rgbColor[2] / 255.0,
+    ],
+    u_threshold: threshold,
+  };
+
+  var fragSrc = `
+        precision lowp float;
+        varying vec2 vTextureCoord;
+        uniform sampler2D uSampler;
+
+        uniform vec3 u_colorToRemove;
+        uniform float u_threshold;
+
+        void main() {
+
+            vec2 coord = vTextureCoord;
+            vec4 texColor = texture2D(uSampler, coord);
+            vec4 colorToRemove = vec4(u_colorToRemove,0.0);		
+
+            vec3 transparent_diff = texColor.rgb - colorToRemove.rgb;
+            float transparent_distance = length(transparent_diff);
+
+            if(transparent_distance < u_threshold)
+                discard;
+
+            gl_FragColor = vec4(vec3(texColor.rgb),texColor.a);
+
+        }
+`
+    .split('\n')
+    .reduce((c, a) => c + a.trim() + '\n');
+
+  var filter = new PIXI.Filter(null, fragSrc, uniforms);
+
+  filter.apply = function (filterManager, input, output) {
+    this.uniforms = uniforms;
+
+    // draw the filter...
+    filterManager.applyFilter(this, input, output);
+  };
+
+  this._renderer.setFilter(filter);
+};
+
 /**
  * @param obj The target object
  * @param scene The scene containing the object
  * @deprecated
  */
-gdjs.SpriteRuntimeObject.prototype.turnTowardObject = function(obj, scene) {
-    if ( obj === null ) return;
-
-    this.rotateTowardPosition(obj.getDrawableX()+obj.getCenterX(),
-        obj.getDrawableY()+obj.getCenterY(), 0, scene);
+gdjs.SpriteRuntimeObject.prototype.turnTowardObject = function (obj, scene) {
+  if (obj === null) return;
+
+  this.rotateTowardPosition(
+    obj.getDrawableX() + obj.getCenterX(),
+    obj.getDrawableY() + obj.getCenterY(),
+    0,
+    scene
+  );
```

Comment:
All your changes here are using PIXI specific code, so they should be implemented in the renderer.

Reply:
Yes, bien vu!